### PR TITLE
UI: Proper explanation for the global setting to avoid ambiguity

### PR DESCRIPTION
**Pull request by Pearl1594**
### Description
Fixes: #10041 
This PR  provides Proper explanation for the global setting to avoid ambiguity

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [X] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [X] Minor

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

File: engine/schema/src/main/resources/META-INF/db/schema-41910to42000.sql [hunk 1]
```diff
@@ -425,3 +425,5 @@ INSERT IGNORE INTO `cloud`.`guest_os_hypervisor` (uuid, hypervisor_type, hypervi
 
 CALL `cloud`.`IDEMPOTENT_ADD_COLUMN`('cloud.vm_instance', 'delete_protection', 'boolean DEFAULT FALSE COMMENT "delete protection for vm" ');
 CALL `cloud`.`IDEMPOTENT_ADD_COLUMN`('cloud.volumes', 'delete_protection', 'boolean DEFAULT FALSE COMMENT "delete protection for volumes" ');
+
+UPDATE `cloud`.`configuration` set name = "kvm.host.auto.enable.disable" where name = "enable.kvm.host.auto.enable.disable";
```
Comment (DaanHoogland) [inline engine/schema/src/main/resources/META-INF/db/schema-41910to42000.sql:429]: 4.20 is released, so this should go somewhere else

**Review by DaanHoogland — COMMENTED** [review]
looks mostly good, needs testing

File: ui/src/views/infra/HostEnableDisable.vue [hunk 2]
```diff
@@ -31,7 +31,7 @@
           <span v-html="$t('message.confirm.enable.host')" />
```
Comment (shwstppr) [inline ui/src/views/infra/HostEnableDisable.vue:31]:
```suggestion
          <span v-html="resourcestate === 'Enabled' ? $t('message.confirm.disable.host') : $t('message.confirm.enable.host')" />
```

**Review by shwstppr — COMMENTED** [review]
@Pearl1594 I think if we just do the following change then it should be good enough. In my opinion, global config is largely okay.

File: ui/src/views/infra/HostEnableDisable.vue (hunk 2 above)
Comment (Pearl1594) [reply ui/src/views/infra/HostEnableDisable.vue:31]: The global config name has a whole lot of enable - disable reference which was ambiguous - which was why the gs name was changed. Do you think that isn't required?

Comment (shwstppr) [reply ui/src/views/infra/HostEnableDisable.vue:31]: @Pearl1594 maybe @nvazquez can comment but I think global setting is largely okay as it allows enabling or disabling a KVM host based on health check results.

Comment (nvazquez) [reply ui/src/views/infra/HostEnableDisable.vue:31]: I agree with @shwstppr, I was personally fine with the setting name but it seems to be a bit confusing for users. What about changing the setting name to: `allow.kvm.host.auto.enable.disable` ?

Comment (Pearl1594) [reply ui/src/views/infra/HostEnableDisable.vue:31]: I like `allow.kvm.host.auto.enable`. I feel the use of disable in the name makes it redundant as it is indicated in the value being set against the global setting. Does that seem alright @nvazquez ?

**Comment by Pearl1594** [comment]
@blueorangutan package

File: ui/src/views/infra/HostEnableDisable.vue [hunk 3]
```diff
@@ -28,15 +28,15 @@
...
         </template>
       </a-alert>
-      <div v-show="enableKVMAutoEnableDisableSetting" class="reason">
+      <div v-show="kvmAutoEnableDisableSetting" class="reason">
         <a-form-item
           class="form__item"
           name="reason"
           ref="reason"
-          :label="'The setting \'enable.kvm.host.auto.enable.disable\' is enabled, ' +
+          :label="'The setting \'kvm.host.auto.enable.disable\' is enabled, ' +
```
Comment (nvazquez) [inline ui/src/views/infra/HostEnableDisable.vue:39]: Maybe we can also rephrase this to something like: `The Auto Enable/Disable KVM Hosts functionality is enabled`

Comment (nvazquez) [reply ui/src/views/infra/HostEnableDisable.vue:39]: Sorry to bother, I have just re-read the original issue and it seems to me we can fix it without needing to update the setting name but just updating this text and also the title as @shwstppr pointed out, leaving the setting as it originally was. What do you think @Pearl1594 @shwstppr ?

Comment (Pearl1594) [reply ui/src/views/infra/HostEnableDisable.vue]: Fine, I've reverted it.

File: engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java [hunk 4]
```diff
@@ -1397,9 +1397,10 @@ private void processHostHealthCheckResult(Boolean hostHealthCheckResult, long ho
                 logger.error("Unable to find host with ID: {}", hostId);
                 return;
             }
+
             if (!BooleanUtils.toBoolean(EnableKVMAutoEnableDisable.valueIn(host.getClusterId()))) {
                 logger.debug("{} is disabled for the cluster {}, cannot process the health check result " +
-                        "received for the host {}", EnableKVMAutoEnableDisable.key(), host.getClusterId(), host);
+                        "received for the host %s", EnableKVMAutoEnableDisable.key(), host.getClusterId(), host.getName());
```
Comment (DaanHoogland) [inline engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java:1403]:
```suggestion
                        "received for the host {}", EnableKVMAutoEnableDisable.key(), host.getClusterId(), host.getName());
```

**Review by shwstppr — COMMENTED** [review]
@Pearl1594 I guess we're in somewhat agreement that changes should be limited to ui/src/views/infra/HostEnableDisable.vue
Can you please revert other changes and rename the PR title?

File: ui/src/views/infra/HostEnableDisable.vue [hunk 5]
```diff
@@ -90,9 +90,9 @@ export default {
       if (this.resource.hypervisor !== 'KVM') {
         return
       }
-      api('listConfigurations', { name: 'enable.kvm.host.auto.enable.disable', clusterid: this.resource.clusterid }).then(json => {
-        if (json.listconfigurationsresponse.configuration[0]) {
-          this.enableKVMAutoEnableDisableSetting = json.listconfigurationsresponse.configuration[0].value
+      api('listConfigurations', { name: 'kvm.host.auto.enable.disable', clusterid: this.resource.clusterid }).then(json => {
```
Comment (shwstppr) [inline ui/src/views/infra/HostEnableDisable.vue:93]: not needed now I guess?

File: engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java [hunk 6]
```diff
@@ -1397,9 +1397,10 @@ private void processHostHealthCheckResult(Boolean hostHealthCheckResult, long ho
                 logger.error("Unable to find host with ID: {}", hostId);
                 return;
             }
+
```
Comment (nvazquez) [inline engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java:1400]: Can we revert changes on this file?

File: server/src/main/java/com/cloud/resource/ResourceManagerImpl.java [hunk 7]
```diff
@@ -1840,9 +1840,9 @@ private void handleAutoEnableDisableKVMHost(boolean autoEnableDisableKVMSetting,
                 _hostDetailsDao.update(hostDetail.getId(), hostDetail);
             } else if (!isUpdateFromHostHealthCheck && hostDetail != null &&
                     Boolean.parseBoolean(hostDetail.getValue()) && resourceEvent == ResourceState.Event.Disable) {
-                logger.info(String.format("The setting %s is enabled but the host %s is manually set into %s state," +
+                logger.info("The setting {} is enabled but the host {} is manually set into {} state," +
```
Comment (nvazquez) [inline server/src/main/java/com/cloud/resource/ResourceManagerImpl.java:1843]: And this file also?

**Review by nvazquez — APPROVED** [review]
Thanks @Pearl1594 LGTM - just a few non-blocking comments to keep the changes to minimal

File: server/src/main/java/com/cloud/resource/ResourceManagerImpl.java (hunk 7 above)
Comment (Pearl1594) [reply server/src/main/java/com/cloud/resource/ResourceManagerImpl.java:1843]: What's the issue with this ? this was being reported by sonar - so I fixed it.

Comment (nvazquez) [reply server/src/main/java/com/cloud/resource/ResourceManagerImpl.java:1843]: Not an issue, I just see it redundant

File: engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java [hunk 8]
```diff
@@ -1399,7 +1399,7 @@ private void processHostHealthCheckResult(Boolean hostHealthCheckResult, long ho
             }
             if (!BooleanUtils.toBoolean(EnableKVMAutoEnableDisable.valueIn(host.getClusterId()))) {
                 logger.debug("{} is disabled for the cluster {}, cannot process the health check result " +
-                        "received for the host {}", EnableKVMAutoEnableDisable.key(), host.getClusterId(), host);
+                        "received for the host {}", EnableKVMAutoEnableDisable.key(), host.getClusterId(), host.getName());
```
Comment (shwstppr) [inline engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java:1402]: I would prefer this without `host.getName()` change as then it would give more details of hosts using 
```suggestion
                        "received for the {}", EnableKVMAutoEnableDisable.key(), host.getClusterId(), host);
```

File: server/src/main/java/com/cloud/resource/ResourceManagerImpl.java (hunk 7 above)
Comment (shwstppr) [inline server/src/main/java/com/cloud/resource/ResourceManagerImpl.java:1843]:
```suggestion
                logger.info("The setting {} is enabled but the {} is manually set into {} state," +
```

File: server/src/main/java/com/cloud/resource/ResourceManagerImpl.java [hunk 9]
```diff
@@ -1840,9 +1840,9 @@ private void handleAutoEnableDisableKVMHost(boolean autoEnableDisableKVMSetting,
...
                                 "ignoring future auto enabling of the host based on health check results",
-                        AgentManager.EnableKVMAutoEnableDisable.key(), host.getName(), resourceEvent));
+                        AgentManager.EnableKVMAutoEnableDisable.key(), host.getName(), resourceEvent);
```
Comment (shwstppr) [inline server/src/main/java/com/cloud/resource/ResourceManagerImpl.java:1845]:
```suggestion
                        AgentManager.EnableKVMAutoEnableDisable.key(), host, resourceEvent);
```

**Review by shwstppr — APPROVED** [review]
minor suggestions, otherwise code lgtm

File: server/src/main/java/com/cloud/resource/ResourceManagerImpl.java (hunk 7 above)
Comment (nvazquez) [reply server/src/main/java/com/cloud/resource/ResourceManagerImpl.java]: I meant using `String.format` or `{}` inside the logger.info has the same effect, anyways I'm fine with it

Comment (Pearl1594) [reply server/src/main/java/com/cloud/resource/ResourceManagerImpl.java]: String.format() has been removed though.

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan test

File: engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java [hunk 10]
```diff
@@ -1399,7 +1399,7 @@ private void processHostHealthCheckResult(Boolean hostHealthCheckResult, long ho
             }
             if (!BooleanUtils.toBoolean(EnableKVMAutoEnableDisable.valueIn(host.getClusterId()))) {
                 logger.debug("{} is disabled for the cluster {}, cannot process the health check result " +
-                        "received for the host {}", EnableKVMAutoEnableDisable.key(), host.getClusterId(), host);
+                        "received for the {}", EnableKVMAutoEnableDisable.key(), host.getClusterId(), host);
```
Comment (DaanHoogland) [inline engine/orchestration/src/main/java/com/cloud/agent/manager/AgentManagerImpl.java:1402]:
```suggestion
                        "received for {}", EnableKVMAutoEnableDisable.key(), host.getClusterId(), host);
```

File: server/src/main/java/com/cloud/resource/ResourceManagerImpl.java [hunk 11]
```diff
@@ -1840,9 +1840,9 @@ private void handleAutoEnableDisableKVMHost(boolean autoEnableDisableKVMSetting,
                 _hostDetailsDao.update(hostDetail.getId(), hostDetail);
             } else if (!isUpdateFromHostHealthCheck && hostDetail != null &&
                     Boolean.parseBoolean(hostDetail.getValue()) && resourceEvent == ResourceState.Event.Disable) {
-                logger.info(String.format("The setting %s is enabled but the host %s is manually set into %s state," +
+                logger.info("The setting {} is enabled but the {} is manually set into {} state," +
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/resource/ResourceManagerImpl.java:1843]:
```suggestion
                logger.info("The setting {} is enabled but {} is manually set into {} state," +
```

**Comment by Pearl1594** [comment]
@blueorangutan package

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by Pearl1594** [comment]
@blueorangutan test
